websocket.go

```diff
@@ -319,15 +319,29 @@ func (c *Client) closeSingleUseConn(dst *network.ServerIdentity, path string) {
 	}
 }
 
-func (c *Client) newConnIfNotExist(dst *network.ServerIdentity, path string) (*websocket.Conn, error) {
+func (c *Client) newConnIfNotExist(dst *network.ServerIdentity, path string) (*websocket.Conn, *sync.Mutex, error) {
 	var err error
 
+	// c.Lock protects the connections and connectionsLock map
+	// c.connectionsLock is held as long as the connection is in use - to avoid that two
+	// processes send data over the same websocket concurrently.
 	dest := destination{dst, path}
 	c.Lock()
-	conn, ok := c.connections[dest]
+	connLock, exists := c.connectionsLock[dest]
+	if !exists {
+		c.connectionsLock[dest] = &sync.Mutex{}
```

Comment:
do we need to delete the entry from the map when the send is done?

Reply:
The current locking needs the `connectionLock` to stay the same when a caller uses the `Client` for the same node/path destination twice in a row. The connection will be closed between the two calls and deleted in `c.connections`, but the lock must stay the same.